### PR TITLE
Update nginx frontend config to support additional clusters.

### PR DESCRIPTION
**Pull request by absoludity**
### Description of the change

Updates the helm chart to allow configuring the nginx frontend to
proxy to additionally configured clusters, including the ca-cert.

Additionally updates the local devel setup to easily add an additional
cluster with oidc options pointing to dex running on the initial
cluster.

### Benefits
Allows requests to additional clusters using the users credential (ie. limited by users configured RBAC on that cluster), as shown:
kubeapps-request-to-additional-cluster

### Applicable issues

Follows #1758 . Ref #1752 .

### Additional information

Unrelated to this change, but I noticed that the request I used to test is something we shouldn't be requesting (it's trying to request secrets in all namespaces, but is using the namespace `_all`). For an actual namespace (`kubeapps`) it succeeds but isn't even necessary (commented on #1686). I'm assuming that this means currently if you have an app repo in the `kubeapps` namespace which happens to have a secret for the app repo (it can't have an image pull secret) then it would fail to load.

### CODE REVIEW

File: chart/kubeapps/templates/kubeapps-frontend-config.yaml [hunk 1]
```diff
@@ -55,6 +57,20 @@ data:
         include "./server_blocks/k8s-api-proxy.conf";
       }
 
+      # Ensure each additional cluster can be reached (should only be
+      # used with an auth-proxy where k8s credentials are never leave
```
Comment (andresmgot) [inline chart/kubeapps/templates/kubeapps-frontend-config.yaml:61]: where k8s credentials ~~are~~ never leave

File: chart/kubeapps/templates/kubeapps-frontend-config.yaml [hunk 2]
```diff
@@ -55,6 +57,20 @@ data:
...
+      # the cluster). See additionalClusters option.
+      {{- range .Values.featureFlags.additionalClusters }}
+      location ~* /api/clusters/{{ .name }} {
+        rewrite /api/clusters/{{ .name }}/(.*) /$1 break;
+        rewrite /api/clusters/{{ .name }} / break;
+        proxy_pass {{ .apiServiceURL }};
+        proxy_ssl_trusted_certificate "./server_blocks/{{ .name }}-ca.pem";
```
Comment (andresmgot) [inline chart/kubeapps/templates/kubeapps-frontend-config.yaml:68]: is the ca-cert mandatory?

File: chart/kubeapps/values.yaml [hunk 3]
```diff
@@ -694,3 +694,8 @@ featureFlags:
...
...VE13TURVek1EQTFORE0xTVZvd0ZURVRNQkVHQTFVRQpBeE1LYTNWaVpYSnVaWFJsY3pDQ0FTSXdEUVlKS29aSWh2Y05BUUVCQlFBRGdnRVBBRENDQVFvQ2dnRUJBTHhzCmxnNVFDS2p1VjIxMHBINFNRMHBaNkRSMytaU1lqaEdQaE1LRGVjZDVkTVZDK0hUV1RGcDNhR00yK01NM0JhaWIKNlNxcE9QaVNVdk5xc2RlOGtVOXRFd3JNVXMzZ1lsdkFNWk5yOHNPWVNXWm9FTE9lVGd3SlJhVkhIOHBxN3huYQowT2U3N3hsSGtvenB1K1NMcHo2UGxKYzdqc1k3L2ExTklyM2dNL0ZhVzlIQks1bW5ZeWl3UEt3SjBMNWFMY1RrCk5aV3VtVUhOZ2hET1BnaWZZRktOTEhONjVoTkFlbGY1dUFQcGVTOXh2SGZISE51OElYeUhJNEI0Qkx2RmdSQjUKM3ZQZmNFbll4UWUya21ueUFKd1hLTTBEeGdYd0Q1OTkzdDIyclJQQitrMzR5VkViWlJlS3VmQWs5Y29TMjFnbApFbVVmRlVpV0lPd0kxWnlFeEY4Q0F3RUFBYU1qTUNFd0RnWURWUjBQQVFIL0JBUURBZ0trTUE4R0ExVWRFd0VCCi93UUZNQU1CQWY4d0RRWUpLb1pJaHZjTkFRRUxCUUFEZ2dFQkFBS2NYdEtYL0FqNUJKQnpOWHRLVVNDOFpmelUKa2FqNlVkbXpGUkJYMUFheG4xcXZrMlBWb3NBaXV2enpHcW5KR2JFRXFBZnFIYUJQaVp0dnhWMmZHMFZlV2J1agpra1NRcDBid0o3WDAwZjIyVmJHRGVENjRQU0VRbEk2Y3VSNFVad0x4ZFpwakppc0k4bVRIbk4zMXFEcnJ4SzYrCjdocm44b0VZbEtDNjVXaGg1ek84cXQ3SWZvZ2ZCN3pYQW5qaFFSdFEwcE9ZUGhLVHdUemd6TGwyS2tRWGxvZmEKdXlNQ2JjSFBvVkVCb1oyb0N3RTZxbHZzdkFIOWo2V2kvbFZOZlE3OWM3YjNSeVE0TUVsUXVaQklGckVidU9uVAp3MEhWTGh3NUVKSWxQdkI4TU9NaFhmY25Hb1ErUU5OYkhBSXhIWFRlNEdOaWExMG4yQm45NXJYYWlNVT0KLS0tLS1FTkQgQ0VSVElGSUNBVEUtLS0tLQo=
```
Comment (andresmgot) [inline chart/kubeapps/values.yaml:701]: I would use some different commented data:

```
name: second-cluster
apiServiceURL: https://second-cluster:6443
certificateAuthorityData: LS0tLS1CRUd....
```

(literally with the ellipsis, to not include the whole cert)

Should we also add a note first saying this doesn't work yet?

Also, you are missing a blank line at the end.

File: chart/kubeapps/templates/kubeapps-frontend-config.yaml (hunk 2 above)
Comment (absoludity) [reply chart/kubeapps/templates/kubeapps-frontend-config.yaml:68]: Good point, it shouldn't be mandatory as some api servers will have publicly signed certs. Done.

File: chart/kubeapps/values.yaml (hunk 3 above)
Comment (absoludity) [reply chart/kubeapps/values.yaml:701]: Done and done :)

**Pull request merged**